### PR TITLE
EMBCESSMOD-1471

### PR DESCRIPTION
**Pull request by MShayOffice**
Review & Save for Step 1 mostly complete, needs GET to reload Profile data after submission.

### CODE REVIEW

File: responders/src/UI/embc-responder/src/app/core/services/global-constants.ts [hunk 1]
```diff
@@ -40,24 +40,39 @@ export const usDefaultObject = {
...
+};
+export const evacueeProfileCreatedMessage = {
+  title: 'Evacuee Profile Saved',
+  text: 'Evacuee profile has been successfully created.'
+};
+export const wizardESSFileMessage = {
+  title: 'Complete all steps',
+  text: 'Please complete all sections of the ESS File prior to submitting.'
+};
+export const lockedStepMessage = {
```
Comment (SodhiA1) [inline responders/src/UI/embc-responder/src/app/core/services/global-constants.ts:65]: locked step message needs to be dynamic. Define different messages for different steps

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/profile-components/profile-review/profile-review.component.html [hunk 2]
```diff
@@ -305,7 +311,10 @@ <h2>Security Questions</h2>
                 }}
               </div>
               <div class="col-8 bold">
-                {{ displayAnswer3 }}
+                {{
+                  this.stepCreateProfileService.securityQuestions[2].answer
```
Comment (SodhiA1) [inline responders/src/UI/embc-responder/src/app/feature-components/wizard/profile-components/profile-review/profile-review.component.html:315]: Use safe navigation operator(?) for all assignments in html

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/profile-components/profile-review/profile-review.component.ts [hunk 3]
```diff
@@ -61,13 +54,48 @@ export class ProfileReviewComponent implements OnInit {
...
     if (this.verifiedProfileFC.valid) {
-      this.stepCreateProfileService.setTabStatus('review', 'complete');
-      this.wizardService.setStepStatus('/ess-wizard/create-ess-file', false);
+      this.saveLoader = true;
+
+      this.evacueeProfileService
+        .upsertProfile(this.stepCreateProfileService.createProfileDTO())
+        .subscribe(
+          (profileId) => {
+            this.cacheService.set('profileId', profileId);
```
Comment (SodhiA1) [inline responders/src/UI/embc-responder/src/app/feature-components/wizard/profile-components/profile-review/profile-review.component.ts:69]: Use a service to set and get profileId

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/profile-components/profile-review/profile-review.component.ts [hunk 4]
```diff
@@ -61,13 +54,48 @@ export class ProfileReviewComponent implements OnInit {
...
+                  '/ess-wizard/create-ess-file/evacuation-details'
+                ]);
+              });
+          },
+          (error) => {
+            this.saveLoader = false;
+
+            this.alertService.setAlert(
+              'danger',
+              error?.title || globalConst.createProfileError
```
Comment (SodhiA1) [inline responders/src/UI/embc-responder/src/app/feature-components/wizard/profile-components/profile-review/profile-review.component.ts:95]: Remove error.title and just use custom error messages on the UI

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/wizard.component.ts [hunk 5]
```diff
@@ -55,7 +55,10 @@ export class WizardComponent implements OnInit {
     if (lockedIndicator) {
       $event.stopPropagation();
       $event.preventDefault();
-      this.openLockedModal(globalConst.lockedStepMessage);
+      this.openLockedModal(
+        globalConst.lockedStepMessage.text,
+        globalConst.lockedStepMessage.title
```
Comment (SodhiA1) [inline responders/src/UI/embc-responder/src/app/feature-components/wizard/wizard.component.ts:60]: Locked models does not have titles in wireframes

File: suppliers/src/UI/embc-supplier/src/app/fire-and-forget/submission/submission.component.html [hunk 6]
```diff
@@ -379,7 +379,7 @@
                     </div>
                   </div>
```
Comment (SodhiA1) [inline suppliers/src/UI/embc-supplier/src/app/fire-and-forget/submission/submission.component.html:380]: Remove supplier file

**Review by SodhiA1 — CHANGES_REQUESTED** [review]

File: responders/src/UI/embc-responder/src/app/feature-components/wizard/profile-components/profile-review/profile-review.component.ts (hunk 3 above)
Comment (MShayOffice) [reply responders/src/UI/embc-responder/src/app/feature-components/wizard/profile-components/profile-review/profile-review.component.ts:69]: I'll add get/set functions for this to **evacuee-profile.service**.

**evacuee-search-results.component**, **wizard.component**, and **person-detail-form.component** are all directly interacting with the cache service as well.

Comment (SodhiA1) [reply responders/src/UI/embc-responder/src/app/feature-components/wizard/profile-components/profile-review/profile-review.component.ts:69]: @MShayOffice  If you are looking for code references try LoadTeamListService or LocationsService not the components that are presently under development

Comment (MShayOffice) [reply responders/src/UI/embc-responder/src/app/feature-components/wizard/profile-components/profile-review/profile-review.component.ts:69]: @SodhiA1 Not looking for references, just pointing out that there are other places where this type of thing has been merged in.

**Review by SodhiA1 — APPROVED** [review]

**Pull request merged**